packages/react-email/build-preview-server.mjs

```diff
@@ -1,24 +1,49 @@
 import { spawn } from 'node:child_process';
 import fs from 'node:fs';
+import path from 'node:path';
+import os from 'node:os';
+import { fileURLToPath } from 'node:url';
 
-const nextBuildProcess = spawn('next', ['build'], {
-  detached: true,
-  stdio: "inherit"
+// Define __filename and __dirname for ES modules
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+// Determine the correct executable based on the operating system
+const isWindows = os.platform() === 'win32';
+const nextExecutable = path.resolve(
+  __dirname,
+  'node_modules',
+  '.bin',
+  isWindows ? 'next.cmd' : 'next'
+);
+
+// Spawn the 'next build' process
+const nextBuildProcess = spawn(nextExecutable, ['build'], {
+  stdio: 'inherit',
 });
 
+// Handle graceful shutdown on SIGINT (Ctrl+C)
 process.on('SIGINT', () => {
   nextBuildProcess.kill('SIGINT');
 });
 
+// Handle process exit
```

Comment:
Can we remove these comments? I don't think they really help a lot, since most of the things here are quite easy to understand just from reading the code

Reply:
agree! I've removed the unnecessary comments